Improved left menu, branding colors

onigoetz opened the pull request:
## Proposed changes

### Left Menu

<img width="252" alt="image" src="https://github.com/user-attachments/assets/48199392-c0c8-4782-be45-c7924721b175" />

#### Dark mode version

<img width="251" alt="image" src="https://github.com/user-attachments/assets/cc59df57-0f7a-40ee-b41d-cd0bcc013925" />


### Branding Color

<img width="975" alt="image" src="https://github.com/user-attachments/assets/1ce7f613-3539-43f3-baf6-6c7ead372da8" />

<img width="1467" alt="image" src="https://github.com/user-attachments/assets/d5f56ca8-ee78-4015-82fc-6a90c0232b07" />



#### Dark mode version

<img width="973" alt="image" src="https://github.com/user-attachments/assets/fe83c859-2077-4404-a066-dac20ea6eae3" />

<img width="1464" alt="image" src="https://github.com/user-attachments/assets/ab88de1e-239a-4bb3-8cba-9f53ba183aef" />




## Checklist
- [ ] Tests are passing
- [ ] New tests have been written
- [ ] Documentation has been updated
- [ ] Translations have been updated
- [ ] Breaking changes have been avoided or documented

ragusa87 commented:
Looks good to me (I checked the black theme only)

SergioMendolia reviewed (changes requested):
I really like the changes but there is no menu on mobile anymore, so it not usable in the current state.
Additionally, but not blocking:
- I would love a little transition effect on the hover of the links so that the background is animated 
- I hate BEM and I think it's an ugly syntax, also I understand the use of style="" for width values, in the book row and I am fine with it, but I am less ok with multiplying the instructions there... I really liked [this article](https://www.keithcirkel.co.uk/css-classes-considered-harmful/)  on the subject (maybe it can inspire something)
- What do you think about using a "standardized" palette for the colors, like https://uchu.style/ for example? then maybe we can also make it selectable by the user in their profile in the future?

onigoetz commented:
> I really like the changes but there is no menu on mobile anymore, so it not usable in the current state. 

Ah that's weird, will definitely fix that

> Additionally, but not blocking:
>     * I would love a little transition effect on the hover of the links so that the background is animated

That's already the case, I will change the animation to be a bit longer to make that more visible

> I understand the use of style="" for width values, in the book row and I am fine with it, but I am less ok with multiplying the instructions there...

Totally agree, I wish to keep those to the minimum, for now I've added them only to places where the style is unique / not reusable.
I've already switched a few previous uses of `style=` to components to reuse them. In the case of tables I didn't really start styling them and will come back to them later (lots of small improvements can be made)

>     * I hate BEM and I think it's an ugly syntax, also  I really liked [this article](https://www.keithcirkel.co.uk/css-classes-considered-harmful/)  on the subject (maybe it can inspire something)

hmm, that can bring a lengthy debate, let's discuss :D. (Sorry for the long answer already 😬 )

That's an interesting article, providing some interesting ideas.

In short, I would propose we check how this attribute thingy works in our use case, to have `.Card[data-size=big]` instead of `.Card--big` (for example). Two arguments against it would be:
1. `data-size=big` could return you many things like buttons or anything else. On the other hand, `Card--big` is very searchable, look for that in your codebase and you're 100% sure you found where it's used.
2. it's longer; that's purely cosmetic

but I have two questions.
1. BEM is ugly, a valid opinion and I share it, but how is that an issue?
2. If we choose something else, what are we trying to solve?

* complexity / verbosity?
* CSS specificity issues?
* class length?
* ease of applying design consistency?

I personally find that BEM does OK on these aspects.

My general reasoning when thinking of CSS and components:
- The number of variants offered should remain low; good components (or actually, good design systems) only offer very little customizations as combinations of customizations are always confusing to read no matter the technique used:
  - `<div class="Card big bold nohover">`
  - `<div class="Card Card--big Card--bold Card--nohover">`
  - `<div class="Card" data-size=big data-text=bold data-animation=nohover>`
  - `<div class="bg-white rounded-lg shadow p-8 w-96 font-bold border-l-4 border-blue-500">`
- I try to think of all my CSS as a component as components are easy to compose: a Title component can be used to be the title of a card or the title of a page. A Menu component can be used for a Dropdown or for the results of a search field)
- A component is not responsible for its own layout: it should take 100% of its available width and should not define margins.

My reasoning for using BEM:
- Uppercase class names: to clearly differentiate from Bootstrap classes (e.g we have a `.Card` and bootstrap has a `.card`)
- BEM works really well with the concept of a components; all classes with the same prefix belong to the same component.

Things I would preferably exclude from possible options: (although not exactly alternatives to the discussion at hand)
- atomic css / utility classes (Tailwind and friends): Has its uses but is a mess to maintain over time and have visual consistency
- custom components: briefly explained in the article, interesting for the state pseudo selector thing but offers no other advantage
- CSS modules: really nice, but requires a compilation step in relation to the UI part (such as React Components) to be correctly integrated
- SASS or other preprocessors: nowadays CSS is crazy good, features that SASS brings to the table is not a crazy improvements compared to what standard CSS can do (yes, nesting is awesome, but that's coming very soon to CSS)

>     * What do you think about using a "standardized" palette for the colors, like https://uchu.style/ for example? then maybe we can also make it selectable by the user in their profile in the future?

The structure I am putting in place with CSS variables will allow us to easily override the default color and create a new theme both for light and dark themes easily.

However there is a big challenge that comes with customizable themes; finding good colors.
We already use three colors, the main color and two lighter colors. add the dark mode and that's 6 colors (as we cannot reuse the same colors).

There are three ways to tackle that:
5. Offer a single color input field: we would need to perform some color math and that's risky as the results are often disappointing.
6.  Offer 6 color input fields for users to fully customize their experience: I doubt anybody would use that feature
7. Offer a set of pre-selected color themes: we could create 5-6 color palettes to propose to our users, a one time chore of finding matching colors.

None of those is fully satisfactory to me but in the end it boils down to one question; who would use this and why?

I would be happy to explore these questions in the future, once we replace some of the places where the theming uses the default Bootstrap colors (Buttons, links...). At this point offering a customization feature would really have an effect on the look-and-feel.

onigoetz commented:
Mobile navigation is back

<img width="642" alt="image" src="https://github.com/user-attachments/assets/a7442547-3e4d-4821-a699-99f7758fba6c" />

<img width="642" alt="image" src="https://github.com/user-attachments/assets/82239661-87b6-4781-ba2a-d78e31cd35ec" />
